src/EventStore.Core/Services/Transport/Grpc/Redaction.SwitchChunk.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Threading.Tasks;
+using EventStore.Client.Redaction;
+using EventStore.Core.Data.Redaction;
+using EventStore.Core.Messages;
+using EventStore.Core.Messaging;
+using EventStore.Plugins.Authorization;
+using Grpc.Core;
+
+namespace EventStore.Core.Services.Transport.Grpc {
+	internal partial class Redaction {
+		private static readonly Operation SwitchChunkOperation = new(EventStore.Plugins.Authorization.Operations.Node.Redaction.SwitchChunk);
+
+		public override async Task SwitchChunk(
+			IAsyncStreamReader<SwitchChunkReq> requestStream,
+			IServerStreamWriter<SwitchChunkResp> responseStream,
+			ServerCallContext context) {
+
+			var user = context.GetHttpContext().User;
+			if (!await _authorizationProvider.CheckAccessAsync(user, SwitchChunkOperation, context.CancellationToken).ConfigureAwait(false))
+				throw RpcExceptions.AccessDenied();
+
+			await SwitchChunksLock().ConfigureAwait(false);
+			try {
+				await SwitchChunks(context, requestStream, responseStream).ConfigureAwait(false);
+			} finally {
+				await SwitchChunksUnlock().ConfigureAwait(false);
+			}
+		}
+
+		private async Task SwitchChunksLock() {
+			var lockTcs = new TaskCompletionSource<RedactionMessage.SwitchChunkLockCompleted>();
```

Comment:
TaskCreationOptions.RunContinuationsAsynchronously

Reply:
:+1:

Reply:
fixed in TcsEnvelope changes: 1bc7a6c4dfc0d37d6db9f204654b89ccbd5487eb